src/Client/MqttLastWill.cs

```diff
@@ -20,18 +24,42 @@ public class MqttLastWill : IEquatable<MqttLastWill>
         /// </param>
         /// <param name="retain">Specifies if the message should be retained or not</param>
         /// <param name="message">Content of the will message to publish</param>
+		[Obsolete ("This constructor is obsolete, please use the constructor with the byte[] payload parameter instead.")]
+		[EditorBrowsable (EditorBrowsableState.Never)]
 		public MqttLastWill (string topic, MqttQualityOfService qualityOfService, bool retain, string message)
 		{
```

Comment:
how about just calling `this(topic, qualityOfService, retain, Encoding.UTF8.GetBytes (message))` instead?